### ISSUE TITLE
Handle non-fully active documents

### ISSUE THREAD

**Issue by marcoscaceres** [WICG/webmonetization#215]
Need to make sure that when documents become non-fully active, the events stop firing.

**Comment by sublimator** [comment]
Can you elaborate on that ?

**Comment by marcoscaceres** [comment]
Yeah, sorry... the definition of [fully active](https://html.spec.whatwg.org/#fully-active) is a bit insane, so let me illustrate.

Basically, imagine you have an iframe. You can "steal" a reference to the "monetization" instance, and then remove the iframe:

```JS
// Same origin-iframe
const iframe = document.querySelector("monetized-iframe");

// Lols! steal a reference. 
const monetization = iframe.contentWindow.navigator.monetization;
monetization.addEventListener("progress", trackProgress);

// We now remove the iframe, which causes it to no longer be "fully active".
iframe.remove();

// The iframe no longer has a window object... it's detached. 
```

So the stolen reference to `monetization` shouldn't receive any more payments, even though it has events registered to it.

**Comment by marcoscaceres** [comment]
Note, this also affects how a browser's "[back/forward cache](https://web.dev/bfcache/)" (BFCache) would work with WM. 

Imagine, I'm looking at a monetized page, and it's receiving payments... now, I hit the back button. I hit the forward button.  

What should happen? If the page is monetized, can the page be put into the "bfcache" or should it bust the cache and be reloaded completely. It's probably ok to cache the page and just keep firing events... but the monetization agent needs to pause and resume.

**Comment by sublimator** [comment]
Cache question aside:

Oh, interesting, has that been a problem in the past with the Coil
Extension? From memory, if you removed the iframe, the associated stream
would be closed and no more events would be routed to that frame/window to
be dispatched on document.monetization. It would be a useless reference.
